### PR TITLE
Added 4 additional modules to the Cellprofiler tool

### PR DESCRIPTION
**Pull request by sunyi000**
- added Tile, ColorToGray, TrackObjects and OverlayOutlines modules 
- fixed flake8 linting issue for all python scripts

### CODE REVIEW

**Comment by sunyi000** [comment]
@beatrizserrano just open the pr first and I'm still planning to add more tests to the new modules next week..

Cheers

**Comment by sunyi000** [comment]
@beatrizserrano @bgruening this is ready for review. 
thanks

**Review by beatrizserrano — COMMENTED** [review]
Thanks a lot, @sunyi000! It has been hard for me to find things to improve here :) Most of my comments are due to the fact that we need users to input the name of images/objects rather than selecting from a drop-down menu. I might have missed some... so bitte double-check :)

File: tools/cellprofiler/tile.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
     </configfiles>
     <inputs>
         <expand macro="input_pipeline_param"/>
-        <param name="input_image" type="text" value="DNA" label="Select an input image" help="Image to be tiled, specified in NameAndTypes" />
+        <param name="input_image" type="text" value="DNA" label="Input an input image" help="Image to be tiled, specified in NameAndTypes" />
```
Comment (beatrizserrano) [inline tools/cellprofiler/tile.xml:56]:
```suggestion
        <param name="input_image" type="text" value="DNA" label="Input the name of an input image" help="Image to be tiled, specified in NameAndTypes" />

```

**Comment by sunyi000** [comment]
@beatrizserrano thanks for the review. I believe i have addressed all your comments, except the cdata

the help text is in xml attribute. CDATA can be used on element but not attribute value..

thanks

File: tools/cellprofiler/color_to_gray.xml [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    <configfiles>
+        <inputs name="input_json"/>
+    </configfiles>
+    <inputs>
+        <expand macro="input_pipeline_param"/>
+        <param name="name_input_image" label="Enter the name of the input image" value="DNA" type="text">
+            <expand macro="text_validator" />
+        </param>
+        <conditional name="con_conversion_method">
+            <param name="conversion_method" type="select" label="Conversion method" help="How do you want to convert the color image? &lt;br/&gt;
```
Comment (beatrizserrano) [inline tools/cellprofiler/color_to_gray.xml:144]: https://github.com/bgruening/galaxytools/blob/master/tools/cellprofiler/identify_primary_objects.xml#L15

File: tools/cellprofiler/overlay_outlines.xml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            <conditional name="con_display_mode">
+                <param name="display_mode" type="select" label="Outline display mode" help="Specify how to display the outline contours around your objects.&lt;br/&gt;
+                &lt;i&gt;Color:&lt;/i&gt; outlines produce a clearer display for images where the cell borders have a high intensity, but take up more space in memory.&lt;br/&gt; &lt;i&gt;Grayscale:&lt;/i&gt; outlines are displayed with either the highest possible intensity or the same intensity as the brightest pixel in the image. ">
+                    <option value="Color">Color</option>
+                    <option value="Grayscale">Grayscale</option>
+                </param>
+                <when value="Color">
+                    <repeat name="rpt_obj_to_display" title="Outline" min="1">
+                        <param name="obj_to_display" type="text" label="Input objects to display" help="Choose the objects whose outlines you would like to display. "/>
```
Comment (beatrizserrano) [inline tools/cellprofiler/overlay_outlines.xml:15]:
```suggestion
                        <param name="obj_to_display" type="text" label="Enter the name of the objects to display" help="Choose the objects whose outlines you would like to display. "/>

there are more like this @sunyi000 
```

File: tools/cellprofiler/tile.xml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        <![CDATA[
+            python '$__tool_directory__/tile.py' -p '$input_pipeline' -i '$input_json'
+            ]]>
+    </command>
+    <configfiles>
+        <inputs name="input_json"/>
+    </configfiles>
+    <inputs>
+        <expand macro="input_pipeline_param"/>
+        <param name="input_image" type="text" value="DNA" label="Input the name of an input image" help="Image to be tiled, specified in NameAndTypes" />
```
Comment (beatrizserrano) [inline tools/cellprofiler/tile.xml:56]: actually we used sth different before:
```suggestion
        <param name="input_image" type="text" value="DNA" label="Enter the name of an input image" help="Image to be tiled, specified in NameAndTypes" />

```

File: tools/cellprofiler/color_to_gray.xml (hunk 2 above)
Comment (sunyi000) [reply tools/cellprofiler/color_to_gray.xml:144]: ah.. ok I will change them

**Review by bgruening — APPROVED** [review]

**Review by beatrizserrano — APPROVED** [review]
This looks great @sunyi000, thanks for the super quick process and adopting all the picky suggestions 'gerne' ;) We are ready to create a super cool workflow now!

**Pull request merged**